src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from "react";
+import { VenueAccessMode } from "types/VenueAcccess";
+import { AnyVenue } from "types/venues";
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { joinVenue } from "utils/url";
+import { SecretPasswordForm } from "../SecretPasswordForm";
+
+interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit: () => void;
+  onPasswordSuccess: () => void;
+}
+
+export const VenueJoinButton: React.FC<VenueJoinButtonProps> = ({
+  venueId,
+  venue,
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, venue?.entrance);
+  }, [venue, venueId]);
+
+  // @debt Handle emails and codes as well.
+  if (venue.access === VenueAccessMode.Password) {
+    return (
+      <div className="secret-password-form-wrapper">
+        <SecretPasswordForm
+          buttonText={
+            venue.config?.landingPageConfig.joinButtonText ?? "Join the party"
```

Comment:
Should this be pulled out into a `const` and shared with the 'join the event' text below?

Also, do we want 'party' to be the default fallback? I think we were aiming to use generic terms such as 'event' across the platform, though can't remember.

Reply:
Yes, this should be a `const`, I'm going to change it.

I believe 'party' was the default state. Since most of the venues are parties and not events. We should bring this up on the daily tomorrow for more info and then add the default terms to `settings.ts`

Reply:
Agreed RE: chatting in standup + adding whatever the 'correct default term' is to `settings.ts`

Reply:
Reminder to get back to this + make the const after standup discussion/etc.

Reply:
Changed locally, will let you know when it's ready for re-review.